Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilder.java

```diff
@@ -176,24 +173,26 @@ protected Element sign(final Assertion assertion, final X509Certificate certific
 
     }
 
-    protected Issuer createIssuer(final CallbackProperties properties) {
-        Issuer issuer;
-        final String format = properties.getAssertionIssuerFormat();
-        if (format != null) {
-            LOG.debug("Setting Assertion Issuer format to: {}", format);
-            final String sIssuer = properties.getIssuer();
-            LOG.debug("Setting Assertion Issuer to: {}", sIssuer);
-            if (isValidNameidFormat(format)) {
-                issuer = OpenSAML2ComponentBuilder.getInstance().createIssuer(format, sIssuer);
+    protected Issuer createIssuer(final CallbackProperties properties, final X509Certificate certificate) {
+        String format = properties.getAssertionIssuerFormat();
+        String sIssuer = properties.getIssuer();
+
+        if (!(StringUtils.isNotBlank(format) && isValidNameidFormat(format))) {
+            format = NhincConstants.AUTH_FRWK_NAME_ID_FORMAT_X509;
+        }
+
+        if (!(StringUtils.isNotBlank(sIssuer) && checkDistinguishedName(sIssuer))) {
+            if (certificate != null) {
+                sIssuer = certificate.getIssuerDN().getName();
```

Comment:
Need to retrieve cert's subject name instead of issuer.  Here is sample code certification.getSubjectX500Principal().getName()